File: mpp/Lab7/javaPb9/networking/src/main/java/ro/mpp2024/protocol/JsonProtocolUtils.java
package ro.mpp2024.protocol;

import ro.mpp2024.DTO.DTOutils;
import ro.mpp2024.model.Arbitru;
import ro.mpp2024.model.Participant;
import ro.mpp2024.model.Rezultat;

import java.util.Map;

public class JsonProtocolUtils {
    public static Response createLoggedResponse(Arbitru arbitru) {
        Response response = new Response();
        response.setTypeJava(ResponseType.LOGGED_ACC);
        response.setTypeCsharp(CharpTranslateType.toNumber(ResponseType.LOGGED_ACC));
        response.setUser(DTOutils.getDTO(arbitru));
        return response;
    }

    public static Response createErrorResponse(String errorMessage) {
        Response response = new Response();
        response.setTypeJava(ResponseType.ERROR);
        response.setTypeCsharp(CharpTranslateType.toNumber(ResponseType.ERROR));
        response.setErrormessage(errorMessage);
        return response;
    }

    public static Response createRezultateResponse(Map<Participant, Long> participants) {
        Response response = new Response();
        response.setTypeJava(ResponseType.REZULTATE);
        response.setTypeCsharp(CharpTranslateType.toNumber(ResponseType.REZULTATE));
        response.setPunctaje(DTOutils.getDTO(participants));
        return response;
    }

    public static Response createAllRezultateResponse(Map<Participant, Long> participants) {
        Response response = new Response();
        response.setTypeJava(ResponseType.ALL_REZULTATE);
        response.setTypeCsharp(CharpTranslateType.toNumber(ResponseType.ALL_REZULTATE));
        response.setPunctaje(DTOutils.getDTO(participants));
        return response;
    }

    public static Response createParticipantiResponse(Iterable<Participant> participanti) {
        Response response = new Response();
        response.setTypeJava(ResponseType.PARTICIPANTI);
        response.setTypeCsharp(CharpTranslateType.toNumber(ResponseType.PARTICIPANTI));
        response.setParticipanti(DTOutils.getDTOParticipanti(participanti));
        return response;
    }

    public static Response createOkResponse() {
        Response response = new Response();
        response.setTypeJava(ResponseType.OK);
        response.setTypeCsharp(CharpTranslateType.toNumber(ResponseType.OK));
        return response;
    }

    public static Response createRezultatAddedResponse(Rezultat rezultat) {
        Response response = new Response();
        response.setTypeJava(ResponseType.REZULTAT_ADDED);
        response.setTypeCsharp(CharpTranslateType.toNumber(ResponseType.REZULTAT_ADDED));
        response.setRezultat(DTOutils.getDTO(rezultat));
        return response;
    }

    public static Request createLoginRequest(Arbitru arbitru) {
        Request request = new Request();
        request.setTypeJava(RequestType.LOGIN);
        request.setTypeCsharp(CharpTranslateType.toNumber(RequestType.LOGIN));
        request.setUser(DTOutils.getDTO(arbitru));
        return request;
    }

    public static Request createAddRezultatRequest(Rezultat rezultat) {
        Request request = new Request();
        request.setTypeJava(RequestType.ADD_REZULTAT);
        request.setTypeCsharp(CharpTranslateType.toNumber(RequestType.ADD_REZULTAT));
        request.setRezultat(DTOutils.getDTO(rezultat));
        return request;
    }

    public static Request createGetRezultateRequest() {
        Request request = new Request();
        request.setTypeJava(RequestType.GET_REZULTATE);
        request.setTypeCsharp(CharpTranslateType.toNumber(RequestType.GET_REZULTATE));
        return request;
    }

    public static Request createGetAllRezultateRequest() {
        Request request = new Request();
        request.setTypeJava(RequestType.GET_ALL_REZULTATE);
        request.setTypeCsharp(CharpTranslateType.toNumber(RequestType.GET_ALL_REZULTATE));
        return request;
    }

    public static Request createGetParticipantiRequest() {
        Request request = new Request();
        request.setTypeJava(RequestType.GET_PARTICIPANTI);
        request.setTypeCsharp(CharpTranslateType.toNumber(RequestType.GET_PARTICIPANTI));
        return request;
    }

    public static Request createLogoutRequest(Arbitru arbitru) {
        Request request = new Request();
        request.setUser(DTOutils.getDTO(arbitru));
        request.setTypeJava(RequestType.LOGOUT);
        request.setTypeCsharp(CharpTranslateType.toNumber(RequestType.LOGOUT));
        return request;
    }
}

File: mpp/Lab10/csharpPb9/networking/protocol/JsonProtocolUtils.cs
using networking.DTO;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace networking.protocol
{
    internal class JsonProtocolUtils
    {
        public static Response createLoggedReposponse(Arbitru user)
        {
            Response response = new Response();
            response.typeCsharp = ResponseType.LOGGED_ACC;
            response.user = DTOutils.getDTO(user);
            return response;
        }

        public static Response createErrorResponse(string errorMessage)
        {
            Response response = new Response();
            response.typeCsharp = ResponseType.ERROR;
            response.errormessage = errorMessage;
            return response;
        }

        public static Response createRezultateResponse(Dictionary<Participant, long> punctajParticipant)
        {
            Response response = new Response();
            response.typeCsharp = ResponseType.REZULTATE;
            response.punctaje = DTOutils.getDTO(punctajParticipant);
            return response;
        }

        public static Response createAllRezultateResponse(Dictionary<Participant, long> punctajParticipant)
        {
            Response response = new Response();
            response.typeCsharp = ResponseType.ALL_REZULTATE;
            response.punctaje = DTOutils.getDTO(punctajParticipant);
            return response;
        }

        public static Response createParticipantiResponse(IEnumerable<Participant> participanti)
        {
            Response response = new Response();
            response.typeCsharp = ResponseType.PARTICIPANTI;
            response.participanti = DTOutils.getDTO(participanti);
            return response;
        }

        public static Response createRezultatAddedResponse(Rezultat rezultat)
        {
            Response response = new Response();
            response.typeCsharp = ResponseType.REZULTAT_ADDED;
            response.rezultat = DTOutils.getDTO(rezultat);
            return response;
        }

        public static Response createOkResponse()
        {
            Response response = new Response();
            response.typeCsharp = ResponseType.OK;
            return response;
        }

        public static Request createLoginRequest(Arbitru user)
        {
            Request request = new Request();
            request.typeCsharp = RequestType.LOGIN;
            request.user = DTOutils.getDTO(user);
            return request;
        }

        public static Request createAddRezultatRequest(Rezultat rezultat)
        {
            Request request = new Request();
            request.typeCsharp = RequestType.ADD_REZULTAT;
            request.rezultat = DTOutils.getDTO(rezultat);
            return request;
        }

        public static Request createGetParticipantiRequest()
        {
            Request request = new Request();
            request.typeCsharp = RequestType.GET_PARTICIPANTI;
            return request;
        }

        public static Request createGetRezultateRequest(Arbitru user)
        {
            Request request = new Request();
            request.user = DTOutils.getDTO(user);
            request.typeCsharp = RequestType.GET_REZULTATE;
            return request;
        }

        public static Request createGetAllRezultateRequest()
        {
            Request request = new Request();
            request.typeCsharp = RequestType.GET_ALL_REZULTATE;
            return request;
        }

        public static Request createLogoutRequest(Arbitru user)
        {
            Request request = new Request();
            request.typeCsharp = RequestType.LOGOUT;
            request.user = DTOutils.getDTO(user);
            return request;
        }
    }
}
